src/rocketmq_client.erl

```diff
@@ -151,23 +169,39 @@ tune_buffer(Sock) ->
         = inet:getopts(Sock, [recbuf, sndbuf]),
     inet:setopts(Sock, [{buffer, max(RecBuf, SndBuf)}]).
 
-get_sock(Servers, undefined) ->
-    try_connect(Servers);
-get_sock(_Servers, Sock) ->
+get_sock(Servers, undefined, SSLOpts) ->
+    try_connect(Servers, SSLOpts);
+get_sock(_Servers, Sock, _SSLOpts) ->
     Sock.
 
-try_connect([]) ->
+try_connect([], _SSLOpts) ->
     error;
-try_connect([{Host, Port} | Servers]) ->
+try_connect([{Host, Port} | Servers], SSLOpts) ->
     case gen_tcp:connect(Host, Port, ?TCPOPTIONS, ?TIMEOUT) of
         {ok, Sock} ->
             tune_buffer(Sock),
             gen_tcp:controlling_process(Sock, self()),
-            Sock;
+            case maybe_get_tls_sock(Sock, SSLOpts) of
+                {error, _} ->
+                    try_connect(Servers, SSLOpts);
```

Comment:
nit, maybe add logging here, which will be helpful

Reply:
Yes, I also think that might be helpful. A warning log would be useful indeed.

Reply:
Fixed